packages/website/src/pages/docs/Internal/SqProject.mdx

```diff
@@ -0,0 +1,174 @@
+- first, the module is parsed
+- then its imports are loaded as additional modules
+- then the imported modules are parsed and their imports are loaded and so on
+- as soon as there's a module that's (1) fully loaded, (2) all of its imports are computed, it will be computed ("simulated")
+- this will trigger the simulation of its parents
+- eventually, the output for the topmost, head module will be produced
+
+This chain of events is internally represented with **actions**. When you add a head, the first action will be dispatched internally, which might lead to other actions when something is loaded or a simulation was completed.
+
+You can expand the "Action log" in the diagram above to see the actions that were dispatched in this example.
+
+## APIs for reacting to project changes
+
+To be notified about `SqProject` state changes, you can use one of two approaches:
+
+1. `addEventListener`
+
+```javascript
+project.addEventListener('state', (event) => {
+    console.log(event.data);
+});
+```
+
+- `state` event will fire on any state changes
+- `action` event will fire when `SqProject` dispatches any internal action
+- `output` event will fire when `SqProject` produces a new ouptut
+
+Don't forget to clean up a listener; e.g., in a React hook:
+
+```javascript
+useEffect(() => {
+    const listener: ProjectEventListener<"action"> = (event) => {
+        console.log(event.data);
+    };
+    project.addEventListener("action", listener);
+    return () => project.removeEventListener("action", listener);
+});
+```
+
+2. `waitForOutput` helper
+
+Event listener mechanism is flexible but verbose, so we also provide this shortcut:
+
+```javascript
+const output = await project.waitForOutput(headName);
+```
+
+## Multi-module example
+
+<br />
+
+<DemoProjectStateViewer
+    headName="main"
+    headModule="mainModule"
+    modules={{
+    mainModule: `import "foo" as foo
+import "bar" as bar
+res = foo.x + bar.x`,
+    foo: `export x = 1`,
+    bar: `export y = 2`,
+    }}
+/>
+
+## Linkers
+
+`SqLinker` type is a pluggable object that describes how imports are resolved.
+
+It allows you to customize the way Squiggle modules are resolved and loaded.
+
+For example, on [Squiggle Hub](https://squigglehub.org), we use a custom linker that fetches `hub:*` modules from the server through GraphQL.
+
+In the command-line `squiggle` script, we load modules from the local filesystem.
+
+For now, [Squiggle Playground](/playground) doesn't have a linker, but in the future we plan to add a way to load modules from Squiggle Hub or other sources.
+
+## Garbage collection
+
+When you add a head to the project, it will start computing the outputs for all modules that are reachable from the head.
+
+When you update a head, the project will clean up all modules that are not reachable from the remaining set of heads.
+
+## Dependency resolving and pinning
```

Comment:
Does this discuss pinning?

Reply:
I reorganized the sections so that this is clearer (moved a few paragraphs to the "Linkers" section)